src/scripts/main.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    const handler = (even) => {
```

Comment:
There is a typo in the parameter name 'even'. It should be 'event' to accurately reflect the event object being passed to the handler function.